### PR TITLE
Update SignalProducer.buffer docs to reflect capacity behavior

### LINKED ISSUE #2069: SignalProducer.buffer documentation refers to an event capacity
The [documentation for `buffer`](https://github.com/ReactiveCocoa/ReactiveCocoa/blob/ad1fd34561d3f2f42e0feb420ace5de321906b78/ReactiveCocoa/Swift/SignalProducer.swift#L92-L106) treats the `capacity` as a limit for _events_ (which is what it used to be), when it’s really a capacity for _values_.

### PR DESCRIPTION
**Pull request by neilpa**
Resolves #2069

### CODE REVIEW

File: ReactiveCocoa/Swift/SignalProducer.swift [hunk 1]
```diff
@@ -89,21 +89,23 @@ public struct SignalProducer<T, E: ErrorType> {
...
-	/// SignalProducer for a signal that will send Events from the buffer.
+	/// Creates a queue for events that replays them when new signals are
+	/// created from the returned producer.
 	///
-	/// When events are put into the returned observer (sink), they will be
-	/// added to the buffer. If the buffer is already at capacity, the earliest
-	/// (oldest) event will be dropped to make room for the new event.
+	/// When values are put into the returned observer (sink), they will be
+	/// added to an internal buffer. If the buffer is already at capacity, 
+	/// the earliest (oldest) value will be dropped to make room for the new
```
Comment (NachoSoto) [inline ReactiveCocoa/Swift/SignalProducer.swift]: :+1:

File: ReactiveCocoa/Swift/SignalProducer.swift [hunk 2]
```diff
@@ -89,21 +89,23 @@ public struct SignalProducer<T, E: ErrorType> {
...
+	/// value.
 	///
 	/// Signals created from the returned producer will stay alive until a
-	/// terminating event is added to the buffer. If the buffer does not contain
+	/// terminating event is added to the queue. If the queue does not contain
 	/// such an event when the Signal is started, all events sent to the
```
Comment (NachoSoto) [inline ReactiveCocoa/Swift/SignalProducer.swift]: Should this say `all values sent...` now?

Comment (neilpa) [inline ReactiveCocoa/Swift/SignalProducer.swift]: Yea, missed that one.

**Comment by neilpa** [comment]
:dash:

**Comment by jspahrsummers** [comment]
:+1:

**Pull request merged**